backend/s3/s3.go

```diff
@@ -13,13 +13,12 @@ import (
 )
 
 const connLimit = 10
-const backendPrefix = "restic"
 
-func s3path(t backend.Type, name string) string {
+func s3path(prefix string, t backend.Type, name string) string {
```

Comment:
Make this function a method on `s3`, so we don't have to explicitly pass the prefix every time.